nautobot/dcim/forms.py

```diff
@@ -2357,6 +2358,13 @@ class InterfaceCreateForm(ComponentCreateForm, InterfaceCommonForm):
             "content_types": Interface._meta.label_lower,
         },
     )
+    role = DynamicModelChoiceField(
+        queryset=Role.objects.all(),
+        required=False,
+        query_params={
+            "content_types": Interface._meta.label_lower,
+        },
+    )
```

Comment:
Should this also be a mixin like we have for the FilterForm?

Reply:
Should we do that since we have optional vs non-optional role fields in different models? Same question with Status.

Edited: could it also be the different `query_params` that each model requires that you have to override the mixin anyways?

Reply:
The other mixins use `self.model._meta.label_lower`

Reply:
https://github.com/nautobot/nautobot/issues/5514

Reply:
Looks like changing this broke CI as `InterfaceCreateForm` doesn't have a `.model` attribute - might be the best fix is to add that though?